File: memories/2024/exercise/scala/day13/src/main/scala/santamarket/model/Receipt.scala
package santamarket.model

class Receipt {
  private var totalPrice: Double = 0.0
  private var totalDiscount: Double = 0.0

  def addProduct(price: Double): Unit = {
    totalPrice += price
  }

  def addDiscount(discount: Double): Unit = {
    totalDiscount += discount
  }

  def getTotalPrice: Double = totalPrice - totalDiscount
}

File: memories/2024/exercise/C#/day13/SantaMarket/Model/Receipt.cs
namespace SantaMarket.Model
{
    public class Receipt
    {
        private double totalPrice = 0.0;
        private double totalDiscount = 0.0;

        public void AddProduct(double price)
        {
            totalPrice += price;
        }

        public void AddDiscount(double discount)
        {
            totalDiscount += discount;
        }

        public double GetTotalPrice()
        {
            return totalPrice - totalDiscount;
        }
    }
}
